include/sea_dsa/DsaColor.hh

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/GraphTraits.h"
+#include "llvm/IR/CallSite.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+
+#include "sea_dsa/CallSite.hh"
+#include "sea_dsa/Global.hh"
+#include "sea_dsa/Graph.hh"
+#include "sea_dsa/GraphTraits.hh"
+
+using namespace sea_dsa;
+using namespace llvm;
+
+class Color {
+private:
+  int m_r, m_g, m_b;
+
+public:
+  Color();
+  Color(int r, int g, int b);
+
+  void randColor();
+  std::string stringColor() const;
+};
+
+enum e_color {WHITE, BLACK, GRAY}; // colors for exploration
+
+using ExplorationMap = DenseMap<const Node *, e_color>;
+using ColorMap = DenseMap<const Node *, Color>;
+using NodeSet = DenseSet<const Node *>;
+
+namespace sea_dsa {
+
+// This class is used to print a dsa graph with colored nodes. The coloring is
+// done by GraphExplorer::colorGraph.
+class ColoredGraph {
+private:
+  NodeSet m_safe;
+  ColorMap m_color;
+  Graph &m_g;
+
+public:
+  ColoredGraph(Graph &g, ColorMap &colorM, NodeSet &safe);
+  sea_dsa::Graph & getGraph();
+  sea_dsa::Graph &getGraph() const;
+
+  std::string getColorNode(const Node *n) const;
+  bool isSafeNode(const Node *n) const;
+
+  typedef sea_dsa::Graph::iterator iterator;
+};
+} // end namespace sea_dsa
+
+class GraphExplorer {
+private:
+  static void
```

Comment:
static functions are uncommon in C++. I can't see what is the reason for exposing them here. 
Are these internal functions used in the implementation? If so, they should be moved to `cc` file. 
If they are public API, they should not be private...

Reply:
This has been moved to seahorn and it won't be a class, it will be documented there.